From d275fa4cdaba295f71a3a6eb0a5df051dc9ee09d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Bernon?= <rbernon@codeweavers.com>
Date: Tue, 2 Dec 2025 13:20:10 +0100
Subject: [PATCH 20/49] win32u: Update the window client surface even with no
 children.

Based on a patch from brett.didemus on Bugzilla, thanks!

Fixes: 257364eb5f671b2fd63507373b0427b519dfa71d
Signed-off-by: sewn <sewn@disroot.org>
---
 dlls/win32u/window.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/dlls/win32u/window.c b/dlls/win32u/window.c
index 4c52bc2..66d270f 100644
--- a/dlls/win32u/window.c
+++ b/dlls/win32u/window.c
@@ -2136,11 +2136,11 @@ static void update_window_client_surfaces( HWND hwnd )
     HWND *children;
     int i;
 
+    update_client_surfaces( hwnd );
+
     if (!(children = list_window_children( hwnd ))) return;
     for (i = 0; children[i]; i++) update_window_client_surfaces( children[i] );
     free( children );
-
-    update_client_surfaces( hwnd );
 }
 
 static HICON get_icon_info( HICON icon, ICONINFO *ii )
-- 
2.52.0

